### PR TITLE
944 streamline secir python code

### LINKED ISSUE #944: Streamline secir python code
In https://github.com/SciCompMod/memilio/tree/main/pycode/memilio-simulation/memilio/simulation, the secir model is still named `secir.py` instead of `osecir.py`. I do not know how much this deviation of the new style is existent in the code itself.

Could you have a look and maybe directly adapt it @HenrZu @MaxBetzDLR ?

### PR DESCRIPTION
**Pull request by MaxBetzDLR**
# Changes and Information

Please **briefly list the changes** (main added features, changed items, or corrected bugs) made:

- Change depreciated module name of secir model in python package
- Improve continuity of code structure and naming
- Update documentation
- Script to check missing/old license
- Check and update current examples

If need be, add additional information and what the reviewer should look out for in particular:

-
-

## Merge Request - Guideline Checklist

Please check our [git workflow](https://github.com/SciCompMod/memilio/wiki/git-workflow). Use the **draft** feature if the Pull Request is not yet ready to review.

### Checks by code author

- [x] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [x] New code adheres to [coding guidelines](https://github.com/SciCompMod/memilio/wiki/Coding-guidelines)
- [x] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] Tests are added for new functionality and a local test run was successful (with and without OpenMP)
- [x] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [x] Proper attention to licenses, especially no new third-party software with conflicting license has been added
- [ ] (For ABM development) Checked [benchmark results](https://github.com/SciCompMod/memilio/wiki/Agent-Based-Model-Development) and ran and posted a local test above from before and after development to ensure performance is monitored.

### Checks by code reviewer(s)

- [ ] Corresponding issue(s) is/are linked and addressed
- [ ] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [ ] Appropriate **unit tests** have been added, CI passes, code coverage and performance is acceptable (did not decrease)
- [ ] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [ ] On merge, add 2-5 lines with the changes (main added features, changed items, or corrected bugs) to the merge-commit-message. This can be taken from the **briefly-list-the-changes** above (best case) or the separate commit messages (worst case).

Closes #944

### CODE REVIEW

File: pycode/examples/simulation/migration.py [hunk 1]
```diff
@@ -167,6 +167,6 @@ def run_mobility_example(plot_results=True):
 if __name__ == "__main__":
     arg_parser = argparse.ArgumentParser(
         'migration',
-        description='Example demonstrating the setup and simulation of a geographically resolved SECIR model with travel.')
+        description='Example demonstrating the setup and simulation of a geographically resolved ODE SECIHURD model with travel.')
```
Comment (HenrZu) [inline pycode/examples/simulation/migration.py:170]:
```suggestion
        description='Example demonstrating the setup and simulation of a geographically resolved ODE SECIHURD model with mobility.')
```

File: pycode/memilio-simulation/memilio/simulation/osecir.cpp [hunk 2]
```diff
@@ -222,20 +225,20 @@ PYBIND11_MODULE(_simulation_secir, m)
...
         py::arg("model"));
 
     pymio::bind_ModelNode<mio::osecir::Model<double>>(m, "ModelNode");
     pymio::bind_SimulationNode<mio::osecir::Simulation<>>(m, "SimulationNode");
     pymio::bind_ModelGraph<mio::osecir::Model<double>>(m, "ModelGraph");
-    pymio::bind_MigrationGraph<Simulation>(m, "MigrationGraph");
+    pymio::bind_MigrationGraph<mio::osecir::Simulation<>>(m, "MigrationGraph");
     pymio::bind_GraphSimulation<MigrationGraph>(m, "MigrationSimulation");
 
     //normally, std::vector is bound to any python iterable, but this doesn't work for move-only elements
```
Comment (HenrZu) [inline pycode/memilio-simulation/memilio/simulation/osecir.cpp:244]: change line 260 to:
// These functions are in general not <del>secir</del> model dependent, only with the current config.

**Review by HenrZu — CHANGES_REQUESTED** [review]
Thank you for these changes :) Some comments and then it can be merged

File: pycode/memilio-simulation/memilio/simulation_test/test_pickle.py [hunk 3]
```diff
@@ -1,3 +1,22 @@
+#############################################################################
+# Copyright (C) 2020-2024 MEmilio
+#
+# Authors:
```
[the remark this answers is not in the document]
Comment (MaxBetzDLR) [reply pycode/memilio-simulation/memilio/simulation_test/test_pickle.py:4]: I dont feel confident to add authors to scripts i had nothing to do with. In general it seems ok not to have an author named for every script.

**Review by HenrZu — APPROVED** [review]

**Pull request merged**